packages/application-shell/src/components/quick-access/index.js

```diff
@@ -121,11 +137,73 @@ export default class QuickAccessTrigger extends React.Component {
     this.setState({ isVisible: false });
   };
 
+  // This function is written with the assumption that a project (including the
+  // existence of a project) never changes without a full page reload.
+  // Otherwise we'd need to
+  // - ensure the response we receive belongs to the current project
+  // - refetch the pim search info when the project key changes
+  updatePimSearchInfo = async () => {
+    // skip when there is no project
+    if (!this.props.project) return;
+
+    const canViewProducts = hasSomePermissions(
+      [permissions.ViewProducts, permissions.ManageProducts],
+      this.props.project.permissions
+    );
+
+    // skip checking when user can't view products anyways
+    if (!canViewProducts) return;
+
+    const isProjectIndexed = await this.props.getPimSearchStatusForProject();
+    this.setState({ isProjectIndexed });
+  };
+
   render() {
     return this.state.isVisible && !this.state.hasError ? (
       <QuickAccessContainer>
-        <QuickAccessModal {...this.props} onClose={this.close} />
+        <QuickAccessModal
+          {...this.props}
+          onClose={this.close}
+          isProjectIndexed={this.state.isProjectIndexed}
+        />
       </QuickAccessContainer>
     ) : null;
   }
 }
+
+export default connect(
+  null,
+  (dispatch, props) => ({
+    getPimSearchStatusForProject: () =>
+      dispatch(
+        // TODO this should be sdkActions.head()
+        // and then we should check whether the response code is
+        // - 200 meaning the project is indexed
+        // - 404 meaning the project is not indexed
+        //
+        // But there is a problem in tne node-sdk client as it tries to
+        // .json()-parse the response to HEAD requests which results in an
+        // error, so we send a regular request for now and limit to no results
+        // instead to keep the payload minimal
```

Comment:
I could not get `head` requests to work due to something inside of `node-sdk` which tries to parse a nonexistent response body. We'd need to dig deeper in case we want to solve this. However things seem to work quite fine while making `POST` requests as a workaround.

Reply:
funny because I ran into the exact same problem
ref: https://github.com/commercetools/nodejs/issues/1022

Reply:
The refactor should be quite manageable and it doesn't seem like there's any action going on upstream. I'd rather get PIM Search support in right away and avoid this branch going stale.

What you want to do once `nodejs` supports `HEAD` requests is to change this to `sdkActions.head`, remove the `payload` and adapt all tests (the `createPimAvailabilityCheckSdkMock` function). You also need to add the `sdkActions.head` function to the `'@commercetools-frontend/sdk` package.

Reply:
Sure, it's fine by me 👌